### PR TITLE
Migrating AccountAction incomeStatement using adapter.

### LINKED ISSUE #179: Migrating AccountAction incomeStatement using adapter.

### PR DESCRIPTION
**Pull request by shafeeq-cse**

### CODE REVIEW

File: src/main/java/org/ideoholic/curium/model/account/service/AccountService.java [hunk 1]
```diff
@@ -1072,10 +1072,10 @@ public void getAllLedgers() {
...
 
-	public boolean getIncomeStatement() {
+	public boolean getIncomeStatement(IncomeStatementDto incomeStatementDto) {
 		
-		String fromDate = DataUtil.dateFromatConversionDash(DataUtil.emptyString(request.getParameter("fromdate")));
-		String toDate = DataUtil.dateFromatConversionDash(DataUtil.emptyString(request.getParameter("todate")));
+		String fromDate = DataUtil.dateFromatConversionDash(DataUtil.emptyString(incomeStatementDto.getFromDate()));
+		String toDate = DataUtil.dateFromatConversionDash(DataUtil.emptyString(incomeStatementDto.getToDate()));
 		
 		if(httpSession.getAttribute(BRANCHID)!=null) {
```
Comment (reehanlalkhan) [inline src/main/java/org/ideoholic/curium/model/account/service/AccountService.java]: Remove `httpSession.getAttribute(BRANCHID)` and add `incomeStatementDto.getBranchId()`

**Review by reehanlalkhan — CHANGES_REQUESTED** [review]
Please remove all references to httpSession from AccountService.java

File: src/main/java/org/ideoholic/curium/model/account/service/AccountService.java (hunk 1 above)
Comment (shafeeq-cse) [reply src/main/java/org/ideoholic/curium/model/account/service/AccountService.java]: Made the changes sir, I have changed the data type in dto class from 'int' to 'Integer' as int cannot accept null.

File: src/main/java/org/ideoholic/curium/model/account/service/AccountService.java [hunk 2]
```diff
@@ -1072,14 +1072,14 @@ public void getAllLedgers() {
...
-		String fromDate = DataUtil.dateFromatConversionDash(DataUtil.emptyString(request.getParameter("fromdate")));
-		String toDate = DataUtil.dateFromatConversionDash(DataUtil.emptyString(request.getParameter("todate")));
+		String fromDate = DataUtil.dateFromatConversionDash(DataUtil.emptyString(incomeStatementDto.getFromDate()));
+		String toDate = DataUtil.dateFromatConversionDash(DataUtil.emptyString(incomeStatementDto.getToDate()));
 		
-		if(httpSession.getAttribute(BRANCHID)!=null) {
+		if(incomeStatementDto.getBranchId()!=null) {
 			
-					int branchId = Integer.parseInt(httpSession.getAttribute(BRANCHID).toString());
+					int branchId = Integer.parseInt(incomeStatementDto.getBranchId().toString());
```
Comment (reehanlalkhan) [inline src/main/java/org/ideoholic/curium/model/account/service/AccountService.java]: Change this to: `incomeStatementDto.getBranchId()`

File: src/main/java/org/ideoholic/curium/model/account/service/AccountService.java [hunk 3]
```diff
@@ -1097,7 +1097,7 @@ public boolean getIncomeStatement() {
 				
 				for (Accountdetails accountDetails : accountsDetails) {
 					
-					List<VoucherEntrytransactions> voucherTransactions = new AccountDAO().getVoucherEntryTransactionsBetweenDates(fromDate, toDate, accountDetails.getAccountdetailsid(), Integer.parseInt(httpSession.getAttribute(BRANCHID).toString()));
+					List<VoucherEntrytransactions> voucherTransactions = new AccountDAO().getVoucherEntryTransactionsBetweenDates(fromDate, toDate, accountDetails.getAccountdetailsid(), Integer.parseInt(incomeStatementDto.getBranchId().toString()));
```
Comment (reehanlalkhan) [inline src/main/java/org/ideoholic/curium/model/account/service/AccountService.java]: Change to: `incomeStatementDto.getBranchId()`

**Review by reehanlalkhan — CHANGES_REQUESTED** [review]

**Review by reehanlalkhan — APPROVED** [review]

**Pull request merged**